src/rm.js

```diff
@@ -91,6 +96,56 @@ function isWriteable(file) {
   return writePermission;
 }
 
+function handleFile(file, options) {
+  if (options.force || isWriteable(file)) {
+    // -f was passed, or file is writable, so it can be removed
+    common.unlinkSync(file);
+  } else {
+    common.error('permission denied: ' + file, { continue: true });
+  }
+}
+
+function handleDirectory(file, options) {
+  if (options.recursive) {
+    // -r was passed, so directory can be removed
+    rmdirSyncRecursive(file, options.force);
+  } else {
+    common.error('path is a directory', { continue: true });
+  }
+}
+
+function handleSymbolicLink(file, options) {
+  var stats;
+  try {
+    stats = fs.statSync(file);
+  } catch (e) {
+    // symlink is bad, so just remove it
```

Comment:
Change this to something like "symlink is broken, so only delete the link, not its contents"